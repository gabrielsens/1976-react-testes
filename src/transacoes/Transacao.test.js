import React from "react";
import { render } from "@testing-library/react";
import Transacao from "./Transacao";
import Transacoes from "./Transacoes";

describe('Componente de transação do extrato', () => {
  it('O Snapshot da lista deve sempre pernanecer a mesma', () => {
    const transacoes = [{
        tipo: '08/09/2020',
        valor: 'saque',
        data: '20.00',
      }, {
        tipo: '00/09/2020',
        valor: 'deposito',
        data: '50.00',
      }, {
        tipo: '10/09/2020',
        valor: 'deposito',
        data: '10.00',
      },{
        tipo: '10/09/2020',
        valor: 'saque',
        data: '10.00',
      },
    ];

    const { container } = render(<Transacoes transacoes={transacoes} />)

    expect(container.firstChild).toMatchSnapshot();
  });

  it('O snapshot do componente deve permanecer sempre o mesmo', () => {
    const { container } = render(<Transacao data="08/09/2020" tipo="saque" valor="20.00" />)

    expect(container.firstChild).toMatchSnapshot();
  });
})