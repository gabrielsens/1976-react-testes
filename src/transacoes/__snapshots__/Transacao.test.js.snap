// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente de transação do extrato O Snapshot da lista deve sempre pernanecer a mesma 1`] = `
<div
  data-testid="transacoes"
>
  <div
    class="transacao-container"
  >
    <p>
      20.00
    </p>
    <p />
    <p>
      R$ 
      saque
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      50.00
    </p>
    <p />
    <p>
      R$ 
      deposito
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      10.00
    </p>
    <p />
    <p>
      R$ 
      deposito
    </p>
  </div>
  <div
    class="transacao-container"
  >
    <p>
      10.00
    </p>
    <p />
    <p>
      R$ 
      saque
    </p>
  </div>
</div>
`;

exports[`Componente de transação do extrato O snapshot do componente deve permanecer sempre o mesmo 1`] = `
<div
  class="transacao-container"
>
  <p>
    08/09/2020
  </p>
  <p>
    saque
  </p>
  <p>
    R$ 
    20.00
  </p>
</div>
`;
